### PR TITLE
fix: Roles based mandatory two factor to check all cases

### PR DESCRIPTION
**Pull request by yash-rajpal**
## Proposed changes (including videos or screenshots)

Currently the role-based mandatory 2FA doesn't check for all conditions properly and just checks if user has some kind of 2FA which leaves a few corner cases to not work as expected.
## Steps to test or reproduce

- Configure role based mandatory 2FA for some user.
- Configure email 2FA for this user.
- Disable email 2FA in settings
- Expect the client to enforce mandatory setting up of TOTP 2FA
## Further comments

[CORE-1508](https://rocketchat.atlassian.net/browse/CORE-1508)

[CORE-1508]: https://rocketchat.atlassian.net/browse/CORE-1508?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved mandatory role-based two-factor authentication to always verify which 2FA methods are available before enforcing setup.

* **Tests**
  * Added end-to-end tests covering enforcement behavior when email and/or TOTP 2FA are enabled or disabled, ensuring correct redirects and UI visibility.

### CODE REVIEW

File: apps/meteor/client/views/hooks/useRequire2faSetup.ts [hunk 1]
```diff
@@ -5,14 +5,26 @@ import { Roles } from '../../stores';
...
 	return Roles.use((state) => {
-		// User is already using 2fa
-		if (!user || user?.services?.totp?.enabled || user?.services?.email2fa?.enabled) {
+		if (!user || !is2FAEnabled) {
 			return false;
 		}
 
 		const mandatoryRole = state.find((role) => !!role.mandatory2fa && user.roles?.includes(role._id));
-		return mandatoryRole !== undefined && tfaEnabled;
+		console.log('mandatoryRole', mandatoryRole);
```
Comment (pierre-lehnen-rc) [inline apps/meteor/client/views/hooks/useRequire2faSetup.ts]:
```suggestion
```
👀

Comment (yash-rajpal) [reply apps/meteor/client/views/hooks/useRequire2faSetup.ts]: Ooops!! Removed.
